contracts/pools/PoolCollection.sol

```diff
@@ -1726,6 +1743,11 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
         view
         returns (AverageRates memory)
     {
+        // if the spot rate is 0, reset the average rates
+        if (!spotRate.isPositive()) {
+            return AverageRates({ blockNumber: 0, rate: zeroFraction112(), invRate: zeroFraction112() });
+        }
```

Comment:
Note that this will not really reset the average rate (and its block number), but rather, just return a "zerofied" average rate.
So if the intention is to actually get the `AverageRates` structure in storage reset to zero, then we should probably apply it separately in each one of the 4 places where this function is called (in one of them, it will not even be called when the spot rate is 0).

Reply:
Correct - the method is supposed to return the "calculated" average rates.
When a storage update is needed, the `_updateAverageRates` method is used and it's called during all relevant places - trade, deposit and withdrawal